API.md

```diff
@@ -355,43 +355,46 @@ public struct HTTPResponseStatus: Equatable, CustomStringConvertible, Expressibl
 }
 
 /// HTTP Methods handled by http_parser.[ch] supports
-public enum HTTPMethod : RawRepresentable {
-    /* be future-proof, http_parser can be upgraded */
-    case other(String)
+public struct HTTPMethod : RawRepresentable {
+
+    public let rawValue: String
+    
+    public init(rawValue: String)
+    public init(_ rawValue: String)
         
-    /* everything that http_parser.[ch] supports */
-    case DELETE
-    case GET
-    case HEAD
-    case POST
-    case PUT
-    case CONNECT
-    case OPTIONS
-    case TRACE
-    case COPY
-    case LOCK
-    case MKCOL
-    case MOVE
-    case PROPFIND
-    case PROPPATCH
-    case SEARCH
-    case UNLOCK
-    case BIND
-    case REBIND
-    case UNBIND
-    case ACL
-    case REPORT
-    case MKACTIVITY
-    case CHECKOUT
-    case MERGE
-    case MSEARCH
-    case NOTIFY
-    case SUBSCRIBE
-    case UNSUBSCRIBE
-    case PATCH
-    case PURGE
-    case MKCALENDAR
-    case LINK
-    case UNLINK
+    /* Constants for everything that http_parser.[ch] supports */
+    public static let delete
```

Comment:
Well, that HTTP methods are always written in uppercase (the struct even stores them as uppercase).

If you want, similar to using 200 instead of .ok or .OK.

As I said, I don't care too much and I can see the desire to use lowercase in Swift, but in this particular case I think uppercase makes more sense.